ddmrp/models/product_product.py

```diff
@@ -10,3 +10,23 @@ class Product(models.Model):
     buffer_ids = fields.One2many(
         comodel_name="stock.buffer", string="Stock Buffers", inverse_name="product_id",
     )
+
+    def action_archive(self):
```

Comment:
There are different ways to archive a product. With product variants, removing a variant attribute on the template will archive the corresponding variant without calling action_archive.
This should be done on create/write

Reply:
Nice catch, fixed now. Can you re-check?